zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/junit/ZeebeIntegrationExtension.java

```diff
@@ -263,10 +270,73 @@ private ApplicationResource asNodeResource(final Object testInstance, final Fiel
     return new ApplicationResource(testInstance, field, field.getAnnotation(TestZeebe.class));
   }
 
-  private Store store(final ExtensionContext extensionContext) {
+  private static Store store(final ExtensionContext extensionContext) {
     return extensionContext.getStore(Namespace.create(ZeebeIntegrationExtension.class));
   }
 
+  private static ZeebeResourceGroup resourceGroup(final ExtensionContext extensionContext) {
+    return store(extensionContext)
+        .getOrComputeIfAbsent(
+            ZeebeResourceGroup.class, key -> new ZeebeResourceGroup(), ZeebeResourceGroup.class);
+  }
+
+  @Override
+  public void afterEach(final ExtensionContext extensionContext) {
+    purgeAllResources(extensionContext);
+  }
+
+  private void purgeAllResources(final ExtensionContext extensionContext) {
+    final var resources = resourceGroup(extensionContext).resources();
+    for (final var resource : resources) {
+      if (!ModifierSupport.isStatic(resource.field()) || !resource.annotation().purgeAfterEach()) {
+        continue;
+      }
+      if (resource instanceof final ClusterResource clusterResource) {
+        purgeResource(clusterResource.cluster().anyGateway());
+      } else if (resource instanceof final ApplicationResource appResource) {
+        if (!(appResource.app() instanceof final TestGateway<?> testGateway)) {
+          // Not all `TestApplication`s and can be used to purge data.
+          // For example, a `TestRestoreApplication` does not even expose the cluster actuator.
+          // We use `TestGateway` as a signal that the application hopefully exposes the cluster
+          // actuator and can actually purge data.
+          throw new IllegalArgumentException(
```

Comment:
💭 It would be good to be able to identify also brokers, because sometimes we test only standalone broker without a gateway, but fair.

🔧 Would `TestStandaloneApplication` work?